cdap-common/src/main/java/io/cdap/cdap/common/utils/TimeBucketHasher.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.common.utils;
+
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Utility class for generating time bucketed hashes
+ */
+public class TimeBucketHasher {
+
+
+  /**
+   * Return timed bucketed hash value. Therefore, for a given hash, result is identical as long as call to this
+   * method has happened within the given window.
+   * This method also uses a jitter based on provided key to reduces likelihood of two time bucket with different keys
```

Comment:
What is the problem with two keys that has the same time bucket? The final hash returned by this method would still be different?

Reply:
if all keys have the same time bucket, then every 15 days, we need to upload all cached artifacts.